gdpr/models.py

```diff
@@ -21,27 +33,106 @@ def create_from_purpose_slug(self, purpose_slug, source_object, issued_at=None,
                 expires_at=issued_at + purpose.expiration_timedelta,
                 tag=tag,
                 purpose_slug=purpose_slug,
-                source_object=source_object
+                source_object=source_object,
+                is_active=True
             )
             for related_object in related_objects or ():
                 legal_reason.related_objects.create(object=related_object)
             return legal_reason
         except KeyError:
             raise KeyError('Purpose with slug {} does not exits'.format(purpose_slug))
 
+    def create_or_update_from_purpose_slug(self, purpose_slug, source_object, issued_at=None, tag=None,
```

Comment:
Podle mě příliš široké rozhraní, měly by existovat pouze 3 metody:

- `create_consent` - vytvoří nebo updatuje Legal reason, related objekty se přepíšou
- `exists_valid_consent` - zjistí jesli existuje aktivní a neexpirovaný Legal reason (to je OK)
- `deactivate_consent` - deaktivuje legal reason (to je OK)

To je vše co jako uživatel knihovny potřebuješ.

Reply:
Vážně? Mě se idea skryté aktualizace moc nelíbí.